jib-core/src/main/java/com/google/cloud/tools/jib/image/ImageReference.java

```diff
@@ -236,12 +237,12 @@ private ImageReference(String registry, String repository, String tag) {
   }
 
   /**
-   * Gets the registry portion of the {@link ImageReference}.
+   * Gets the registry host of the {@link ImageReference}.
```

Comment:
What's the consequences of this? I hope `jib:dockerBuild` will push `docker.io/some/name` instead of `registry-1.docker.io/some/name`.

Also, should `RegistryAliasGroup.getHost()` be done outside of this? The getters of this class simply returns its fields as-is, and we may want to keep it that way, as an immutable class and a pure representation of the parsed reference.

Reply:
That is a good question!  I did this so that [`ImageReference#toString()`](https://github.com/GoogleContainerTools/jib/blob/1801d0bcece855a117e22bde93004c051a369b68/jib-core/src/main/java/com/google/cloud/tools/jib/image/ImageReference.java#L319) would preserve the `docker.io` prefix.

With this change, `jib:dockerBuild -Dimage=docker.io/foo/bar/baz` records the image as `foo/bar/baz`.  But `jib:buildTar -Dimage=docker.io/foo/bar/baz` records the `repoTag` as `docker.io/foo/bar/baz`.

But if I do a `docker build -t docker.io/foo/bar/baz .`, docker reports the built image without the `docker.io` prefix.  And the result of `docker save docker.io/foo/bar/baz` doesn't have `docker.io` either.

I'll move this remapping into `parse()`.